packages/common-all/src/dnode.ts

```diff
@@ -1170,23 +1193,80 @@ export class NoteUtils {
     return notesMap;
   }
 
+  /** @deprecated use {@link NoteUtils.colorV2} */
+  static color(opts: { fname: string; vault?: DVault; notes: NotePropsDict }): {
+    color: string;
+    type: "configured" | "generated";
+  } {
+    const ancestors = NoteUtils.ancestors({ ...opts, includeSelf: true });
+    for (const note of ancestors) {
+      if (note.color) return { color: note.color, type: "configured" };
+    }
+    return { color: randomColor(opts.fname), type: "generated" };
+  }
+
   /** Generate a random color for `note`, but allow the user to override that color selection.
    *
    * @param note The fname of note that you want to get the color of.
    * @param notes: All notes in `engine.notes`, used to check the ancestors of `note`.
    * @returns The color, and whether this color was randomly generated or explicitly defined.
    */
-  static color(opts: { fname: string; vault?: DVault; notes: NotePropsDict }): {
+  static colorV2(opts: {
```

Comment:
It looks like there are only 2 references to the original `color` method.  Can we go ahead and update those to avoid creating a `colorv2`?

Reply:
The remaining usages are in the UI, and the UI doesn't have access to `DEngineClient` but a subset of it. Switching it over is possible but it's more involved than just changing the name, so I'd rather leave it for later.

Reply:
the only usage here is in [[../packages/common-frontend/src/utils/tree-view.tsx]] correct? if that is the only usage, would be better to remove it and simplify the code and re-introduce the feature later

we'll want to expose an engine object in the web version as well so we can use a common API regardless where engine methods are being called from. can be added as a backlog task